sampleSpecs/orderedMap.smithy

```diff
@@ -0,0 +1,11 @@
+$version: "1"
+
+namespace smithy4s.example
+
+// TODO: since @preserveKeyOrder changes the underlying type depending on scala version, the bootstrap module codegen task
+// should generate separate srcX_xx directories to render these differences
+// @alloy#preserveKeyOrder
+map OrderedMap {
```

Comment:
```suggestion
@alloy#preserveKeyOrder
map OrderedMap {
```

We can put an issue in for the todo there I think

Reply:
Created https://github.com/disneystreaming/smithy4s/issues/1830 for this as well as a ticket in our internal tracker.

Reply:
So the trait can't be uncommented since the bootstrap module is cross-built across Scala 2.12, 2.13 and 3, and since the codegen uses scala 2.13 for the codegen the OrderedMap is a `SeqMap` which fails to compile for Scala 2.12.